doc/amazon_sagemaker_debugger.rst

```diff
@@ -0,0 +1,356 @@
+.. sectnum::
+
+#########################
+Amazon SageMaker Debugger
+#########################
+
+
+Amazon SageMaker Debugger allows you to detect anomalies while training your machine learning model by emitting relevant data during training, storing the data and thereafter analyzing it.
+
+.. contents::
+
+Background
+==========
+
+Amazon SageMaker provides every developer and data scientist with the ability to build, train, and deploy machine learning models quickly. Amazon SageMaker is a fully-managed service that encompasses the entire machine learning workflow. You can label and prepare your data, choose an algorithm, train a model, and then tune and optimize it for deployment. You can deploy your models to production with Amazon SageMaker to make predictions at lower costs than was previously possible.
+
+SageMaker Debugger provides a way to hook into the training process and emit debug artifacts (*aka* "tensors") that are representative of the state of the training at any given point in the training lifecycle. Debugger then stores the data in real time and provides a way for you to analyze the data through a concept called "rule". Rules encapsulate logic to analyze tensors and react to anomalies therein.
+
+Setup
+=====
+
+To get started, you must satisfy the following prerequisites:
+
+* Specify an AWS Region where you'll train your model.
+* Create an IAM role ARN that is used to give Amazon SageMaker access to your data in Amazon Simple Storage Service (Amazon S3) necessary for training the model. See the `AWS IAM documentation <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html>`__ for how to fine tune the permissions needed.
+
+Capture real-time debugging data during model training in Amazon SageMaker
+==========================================================================
+
+To enable data emission for debugging model training, Amazon SageMaker will initialize a "hook" which, as the name suggests, would attach itself to the training process and emit the data necessary for debugging. To provide the hook's configuration, specify the new option called ``DebuggerHookConfig`` when training your model. For more about the ``DebuggerHookConfig`` object, see the `API documentation <https://sagemaker.readthedocs.io/en/stable/debugger.html#sagemaker.debugger.DebuggerHookConfig>`__.
+
+The ``DebuggerHookConfig`` accepts one or more objects of type ``CollectionConfig``. ``CollectionConfig`` defines the configuration around a tensor collection you intend to emit and save during model training. Collection object helps group tensors for easier handling.
+    as analysis
+
+.. code:: python
+
+    from sagemaker.debugger import CollectionConfig, DebuggerHookConfig
+
+    collection_config = CollectionConfig(
+        name='collection_name',
+        parameters={
+            "key": "value"
```

Comment:
be consistent with single vs double quotes

Reply:
Stuck to single quotes in the entire document.